tests/network/localnet/liblocalnet.py

```diff
@@ -1,15 +1,65 @@
+from collections.abc import Generator
+from contextlib import contextmanager
+
+import tests.network.libs.nodenetworkconfigurationpolicy as libnncp
 from libs.net import netattachdef
-from libs.net.vmspec import add_network_interface, add_volume_disk
+from libs.net.traffic_generator import Client, Server
+from libs.net.vmspec import IP_ADDRESS, add_network_interface, add_volume_disk, lookup_iface_status
 from libs.vm.affinity import new_pod_anti_affinity
 from libs.vm.factory import base_vmspec, fedora_vm
 from libs.vm.spec import CloudInitNoCloud, Interface, Metadata, Multus, Network
 from libs.vm.vm import BaseVirtualMachine, cloudinitdisk_storage
 from tests.network.libs import cloudinit
+from tests.network.libs.nodenetworkconfigurationpolicy import Resource
+
+_IPERF_SERVER_PORT = 5201
+NNCP_INTERFACE_TYPE_OVS_BRIDGE = "ovs-bridge"
+
+
+@contextmanager
+def running_localnet_vms(
```

Comment:
I do not understand this new contextmanager. What resource is it managing exactly?

It could be a simple helper that starts and waits for Fedora VMs to run the guest OS in parallel. (not a contextmanager)
The name should reflect that as well, e.g. `run_vms()` (it has nothing to do with the fact they are localnet ones).